File: python/src/models/request.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float, Enum
from enums import RequestStatusEnum, DonorRhEnum
from sqlalchemy.orm import relationship
from database import Base


class Request(Base):
    __tablename__ = "request"
    id_request = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
    status = Column(Enum(RequestStatusEnum))
    blood_type = Column(String(5), nullable=False)
    Rh = Column(Enum(DonorRhEnum), nullable=False)
    receiving_person = Column(String(60))
    quantity = Column(Float)
    id_doctor = Column(Integer, ForeignKey("doctor.id_doctor", ondelete="CASCADE"))
    id_center = Column(Integer, ForeignKey("transfusion_center.id_center", ondelete="CASCADE"))
    transfusion_center = relationship("TransfusionCenter", back_populates="requests")
    donations = relationship("Donation", back_populates="request")
    doctor = relationship("Doctor", back_populates="requests")

File: nodejs/database/models/request.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const request = sequelize.define('request', {
        id_request: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        status: {
            type: DataTypes.ENUM(['Donation', 'Processing', 'Testing', 'Distribution']),
        },
        blood_type: {
            type: DataTypes.STRING(5),
        },
        Rh: {
            type: DataTypes.ENUM(['Positive', 'Negative']),
        },
        receiving_person: {
            type: DataTypes.STRING(60),
            allowNull: true
        },
        quantity: {
            type: DataTypes.FLOAT,
            allowNull: true
        },
        id_doctor: {
            type: DataTypes.INTEGER
        },
        id_center: {
            type: DataTypes.INTEGER
        }
    }, {
        freezeTableName: true,
        timestamps: false,
        underscored: true,
        name: {
            singular: 'request',
            plural: 'request'
        }
    });

    request.associate = models => {
        request.belongsTo(models.doctor, {
            foreignKey: 'id_doctor'
        });

        request.belongsTo(models.transfusion_center, {
            foreignKey: 'id_center'
        });

        request.hasMany(models.donation, {
            foreignKey: 'id_request',
            onDelete: 'cascade'
        })
    };

    return request;
};
